main.py

```
@@ -53,219 +87,1126 @@
+    </html>
+    """
+    return HTMLResponse(content=html_content)
 
-def initialize_api_agent():
-    """Initialize the agent for API use with proper error handling"""
-    global api_agent
+# Priority Endpoint 1: AnalysePosts
+@app.post("/AnalysePosts", response_model=AnalysisResponse)
+async def analyse_posts(
+    text: Optional[str] = Form(None),
+    files: List[UploadFile] = File(default=[]),
```

Ruff (0.12.2)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)